server/service/appconfig.go

```diff
@@ -416,6 +416,9 @@ func (svc *Service) ModifyAppConfig(ctx context.Context, p []byte, applyOpts fle
 	// 1. To get the JSON value from the database
 	// 2. To update fields with the incoming values
 	if newAppConfig.MDM.EnableDiskEncryption.Valid {
+		if svc.config.Server.PrivateKey == "" {
```

Comment:
Should we allow disable it if `svc.config.Server.PrivateKey == ""` (removed check below uses `mdm.EnableDiskEncryption.Value && mdm.EnableDiskEncryption.Value != oldMdm.EnableDiskEncryption.Value`).